### PR TITLE
Refactor datapackerdataloader to be modular

### PR DESCRIPTION
**Pull request by foreverlms**
## Summary

Refactors the training data layer from the monolithic `DataPackerDataLoader` /
`DataPacker` / `PackingIterableDataset` into a modular, four-role abstraction
wired by a single loader. Behavior is preserved (golden-batch byte-identical to
the legacy loader; resume validated live), and all existing recipes are migrated.

DataDistributor  →  RawItemProcessor  →  SampleBatcher  →  BatchCollator
(shard/shuffle/      (raw item → one     (samples →        (group → one
 resume)             sample dict)        batch groups)     batch dict)

Each role is a small ABC with one required method; pick a built-in per slot or
write your own. `CosmosDataLoader` is a `torch.utils.data.DataLoader` subclass, so
it drops into the existing training loop.

### New dataflow package — `cosmos_framework/data/vfm/dataflow/`
- **Loaders:** `CosmosDataLoader` (+ `batch_size=` sugar → `SimpleBatcher` +
  `DefaultBatchCollator`), `JointCosmosDataLoader` (ratio-weighted heterogeneous join).
- **Distributors:** `IterableDistributor`, `MapDistributor` (resumable),
  `RankPartitionedDistributor`, `MixtureDistributor`.
- **Processors:** `IdentityProcessor` (+ recipe-specific `VLMProcessor`,
  `VideoPhy2Processor`).
- **Batchers:** `SimpleBatcher`, `PoolPackingBatcher`, `SequentialPackingBatcher`.
- **Collators:** `DefaultBatchCollator`, `VFMListCollator` (+ recipe `VLMCollator`).

### Legacy removal
- Deleted `data_packer.py`, `data_packer_dataloader.py`,
  `packing_iterable_dataset.py`, `test_dp_state_distributed.py` (+ old tests).

### Experiment migrations
- VLM `llava_ov` (renamed from `llava_ov_datapacker`, streaming `IterableDistributor`).
- VLM `videophy2_sft_nano`.
- VFM: existing path unchanged; added `vision_sft_nano_v2` (new-loader variant).
- Added `llava_ov_mapresume` — map-style (`load_dataset(streaming=False)` +
  `MapDistributor`) resumable example.

### Config / TOML
- `PATH_REMAPS["vlm"]`: route `dataloader_train.{max_samples_per_batch,
  max_sequence_length}` → nested `batcher.{max_batch_size, max_tokens}`.

### Checkpoint / resume
- Renamed the resume-state selector value `"data_packer"` → `"cosmos_dataloader"`
  and env prefix `DP_STATE_` → `COSMOS_DL_STATE_` (`DataLoaderStateCallback`,
  `JointDataLoaderStateCallback`, `MapDistributor`). On-disk format unchanged.

### CI / tests / docs
- Updated `tests/launch_regression_test.py` + launch scripts for the `llava_ov`
  rename (golden loss keyed by `llava_ov`; workflow `-k llava_ov`).
- Added golden-batch, resume, and per-role unit tests.
- Replaced `docs/custom_dataset.md` with the `CosmosDataLoader` tutorial; removed
  `docs/dataflow.md`.

## Validation

- **Golden-batch equality:** VLM / videophy2 / VFM batches byte-identical to the
  legacy loader.
- **Live save→stop→resume** on `pre_exp012_llava_ov_mapresume` (8 dp ranks,
  `save_iter=100`): per-rank `input_ids` shapes identical across the resume
  boundary — **792 `(iter, rank)` keys, 0 mismatches** — and loss curves match.
  No duplicated/skipped samples on any rank.
- **No CI risk:** the `llava_ov` golden recipe and its streaming data path are
  unchanged; the remap only affects the 3 VLM TOMLs, all of which compose cleanly
  onto a real `PoolPackingBatcher`.

## Hard invariant

Dataloader resume + checkpoint saving must not regress. Held: resume is preserved
through the existing `DataLoaderStateCallback`, with map-style fast-forward and the
multi-sample contiguity guard, and validated end-to-end above.

### CODE REVIEW

File: cosmos_framework/configs/base/experiment/sft/vision_sft_nano_v2.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (lfengad) [inline cosmos_framework/configs/base/experiment/sft/vision_sft_nano_v2.py]: This experiment is mostly the same with the previous one in parts except data loader? could we kind of share the other configs?

File: examples/toml/sft_config/llava_ov_mapstyle_dataloader.toml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# pre_exp012_llava_ov_mapresume — map-style resumable VLM recipe
+# Base config = cosmos_framework/configs/base/vlm/config.py (task="vlm").
```
Comment (lfengad) [inline examples/toml/sft_config/llava_ov_mapstyle_dataloader.toml:5]: the _mapresume name seems could be changed to a more clear one?

File: cosmos_framework/data/vfm/dataflow/batchers_test.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (lfengad) [inline cosmos_framework/data/vfm/dataflow/batchers_test.py]: Some of these tests could be deleted and they seems too fine-grained?

Comment (foreverlms) [reply cosmos_framework/data/vfm/dataflow/batchers_test.py]: yeah, will do this.

File: cosmos_framework/configs/base/experiment/sft/vision_sft_nano_v2.py (hunk 1 above)
Comment (foreverlms) [reply cosmos_framework/configs/base/experiment/sft/vision_sft_nano_v2.py]: will do it

File: cosmos_framework/data/vfm/dataflow/batchers_test.py (hunk 3 above)
Comment (foreverlms) [reply cosmos_framework/data/vfm/dataflow/batchers_test.py]: got it.

File: examples/toml/sft_config/llava_ov_mapstyle_dataloader.toml (hunk 2 above)
Comment (foreverlms) [reply examples/toml/sft_config/llava_ov_mapstyle_dataloader.toml:5]: has been changed to _mapsyle_dataloader

File: examples/toml/sft_config/vision_sft_nano_mapstyle_dataloader.toml [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: OpenMDW-1.1
```
Comment (lfengad) [inline examples/toml/sft_config/vision_sft_nano_mapstyle_dataloader.toml:2]: Could we change the _v2 to the similar name with the suffix of "_mapstyle_dataloder" to make the naming more consistent?

File: cosmos_framework/configs/base/vlm/experiment/llava_ov_vlm.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (lfengad) [inline cosmos_framework/configs/base/vlm/experiment/llava_ov_vlm.py:1]: Similarly, we could remove the "_experiment" suffix to make the naming consistent with other files in the folder

**Review by yy-code-nv — DISMISSED** [review]

File: cosmos_framework/configs/base/vlm/experiment/llava_ov_vlm.py (hunk 5 above)
Comment (foreverlms) [reply cosmos_framework/configs/base/vlm/experiment/llava_ov_vlm.py:1]: good catch, will do it later

**Review by lfengad — DISMISSED** [review]

**Review by lfengad — DISMISSED** [review]

**Review by lfengad — APPROVED** [review]

**Review by yy-code-nv — APPROVED** [review]
